front/app/containers/Admin/pagesAndMenu/containers/CustomPages/Edit/Settings/index.tsx

```diff
@@ -10,8 +10,8 @@ const EditCustomPageSettings = () => {
   const { customPageId } = useParams() as { customPageId: string };
   const customPage = useCustomPage(customPageId);
 
-  const handleOnSubmit = async (formValues: FormValues) => {
-    await updateCustomPage(customPageId, formValues);
+  const handleOnSubmit = (formValues: FormValues) => {
+    updateCustomPage(customPageId, formValues);
   };
```

Comment:
Also deleteed this async/await that had no purpose.

Reply:
But without `await`, when `handleOnSubmit` is run, it just calls `updateCustomPage` not waiting for it to finish. We could also return from `handleOnSubmit` the value returned by `updateCustomPage`. It would also work, but I believe my first approach is fine too.

Reply:
> But without await, when handleOnSubmit is run, it just calls updateCustomPage not waiting for it to finish. It doesn't work.

Did you try it? It works fine without async/await here in the edit form because we're not awaiting anything.

> We could also return from handleOnSubmit the value returned by updateCustomPage. It would also work, but I believe my first approach is fine too.

Not sure what you mean here.

The most important is the change to the onSubmit type. Without specifying it can return a promise, there was a warning in onFormSubmit that the async/await wasn't doing anything. You can remove my code to see it.

Reply:
In the new custom page form on the other hand, we're awaiting the data of the new custom page so we can pass the page id to the router.

Reply:
@brentguf 

> Did you try it?

Yes, it didn't work. 

> It works fine without async/await here in the edit form because we're not awaiting anything.

Do you mean the old edit page form? It has `await Promise.all` https://github.com/CitizenLabDotCo/citizenlab/blob/23cb853b88a2345b80b6a0b3e0a14f4073c69cde/front/app/containers/Admin/pagesAndMenu/containers/EditPageForm/index.tsx#L61

> The most important is the change to the onSubmit type.

Yes, I've left the type https://github.com/CitizenLabDotCo/citizenlab/pull/2695/commits/23cb853b88a2345b80b6a0b3e0a14f4073c69cde

Reply:
> Yes, it didn't work.

It works fine for me.

> It works fine without async/await here in the edit form because we're not awaiting anything.

Nope, I mean `front/app/containers/Admin/pagesAndMenu/containers/CustomPages/Edit/Settings/index.tsx`

Let's leave it for now and do a call if you need more explanation.

Reply:
> It works fine for me.

I've just git-checked-out to this commit 5a40524c69, and it doesn't work for me. The steps:

1. I copy a slug from a random page
2. I enter this slug into the input on a form of another page
3. I submit the form
4. HTTP request returns "422 Unprocessable Entity"

Result: I see a green mark indicating the page was saved.
Expected result: I see an error indicating the page cannot be saved.

Reply:
It would be great to have a call because I don't understand how it can work without await.

Reply:
You're right! In this case it doesn't work. Very good catch. We need a unit test for this (trying to save the slug with an existing slug).